### ISSUE TITLE
Feature: Removing items from waitlist

### ISSUE THREAD

**Issue by marc-vdm** [cauldron/bw_matchbox#57]
It's currently not possible to remove items from a waitlist and waitlisted items don't show `waitlisted` once re-opened. 

If the waitlist button could be toggle-able, that'd be great!

**Comment by cmutel** [comment]
This is an important UX improvement. Currently the `waitlist` attribute is only removed when a [proxy is created](https://github.com/cauldron/bw_matchbox/blob/08abad0a9966e95b4f187637e4f7ee37f46a09ba/bw_matchbox/webapp.py#L731C13-L731C21) (and also via `multi-match`, but that is a special case of the same thing). However, this could easily be added with the `add-attribute` API call, e.g. `add-attribute/<process id>?attr=waitlist&value=0`.

- [ ] When the waitlist button is pressed, the button class `active` state is removed, and the text is changed to `Remove from waitlist`
- [ ] When the `process_detail` template is loaded, and the process in question has the attribute `waitlist=True`, display the button text `Remove from waitlist`
- [ ] Clicking on `Remove from waitlist` changes the button text back to `Waitlist` and the button class to active
- [ ] Clicking on `Remove from waitlist` issues an API call to `add-attribute/<process id>?attr=waitlist&value=0`

**Comment by lilliputten** [comment]
Doing it this way now (on the client for `sharedData`):

```
   // ...
    waitlist: {{ 'true' if ds.get('waitlist') else 'false' }},
```

Is it ok?

**Comment by cmutel** [comment]
> Is it ok?

LGTM

**Comment by lilliputten** [comment]
I see that attrubute (`ds.get('waitlist')`) doesn't update after current requests.

Requests are:

- `?attr=match_type&value=1` (Is it correct?)
- `?attr=waitlist_comment&value=<comment text>`

Is it correct? Probably we should do:

- `?attr=whitelist&value={1|0}`
- `?attr=waitlist_comment&value=<comment text>`

Should we send `waitlist_comment` request with `whitelist&value=0`?

**Comment by cmutel** [comment]
> Requests are:
> ?attr=match_type&value=1 (Is it correct?)
> ?attr=waitlist_comment&value=<comment text>

Yes, that is right.

> ?attr=waitlist_comment&value=<comment text>

Only if there is an actual comment text, but it can also be an empty string.

> We should not show comment dialog for an 'un-waitlist' operation?

Yes, that is correct. One click and done.

**Comment by lilliputten** [comment]
I don't understand. This --

> ?attr=match_type&value=1 (Is it correct?)

-- doesn' work for me (waitlist attribute doesnt' update).

It works if I use `waitlist&...`.

I'll make PR. Take look, pls (at this code for `doMarkWaitlist`, and refactored -- for `markMatched` -- using calls one-by-one there).

Put the next changes here, pls, if smth is wrong there?

**Comment by lilliputten** [comment]
Fixed in PR #77 -- needs to be checked.

**Comment by cmutel** [comment]
> I don't understand. This --
> 
>     ?attr=match_type&value=1 (Is it correct?)
> 
> -- doesn' work for me (waitlist attribute doesnt' update).
> 
> It works if I use waitlist&....

Sorry about that, it should obviously be `waitlist`, `match_type` is just a bug.
